logstash-core/api/lib/app/service.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+require "logstash/instrument/collector"
+
+class LogStash::Api::Service
+
+  include Singleton
+
+  def initialize
+    @snapshot_rotation_mutex = Mutex.new
+    @snapshot = nil
+
+    LogStash::Instrument::Collector.instance.add_observer(self)
+  end
+
+  def stop
+    LogStash::Instrument::Collector.instance.delete_observer(self)
+  end
+
+  def update(time, snapshot)
+    if snapshot_rotation_mutex.try_lock
+      @snapshot = snapshot
+      @snapshot_rotation_mutex.unlock
+    end
+  end
+
+  def get(key=:service_metrics)
+    @snapshot.to_event.to_hash
+  rescue
+    {}
+  end
+end
```

Comment:
we could, but this interface is going to change as for example makes no sense to have ```to_event.to_hash``` to get data out of the snapshot :+1: We can narrow down the possible errors on the fly with the different interactions.